table/tables/tables.go

```diff
@@ -1481,18 +1481,27 @@ func CheckHandleOrUniqueKeyExistForUpdateIgnoreOrInsertOnDupIgnore(ctx context.C
 
 	// Check unique key exists.
 	{
-		for _, idx := range t.Indices() {
-			if !IsIndexWritable(idx) {
-				continue
+		canSkipIgnoreCheck := func(idx table.Index) bool {
+			if !IsIndexWritable(idx) || !idx.Meta().Unique || (t.Meta().IsCommonHandle && idx.Meta().Primary) {
```

Comment:
Should we skip `t.Meta().PKIsHandle && idx.Meta().Primary` too?

Reply:
i think we can skip that because 

https://github.com/pingcap/tidb/pull/23894/files#diff-ccb2771ef1931990c0d0ce3703def7dd53bd21d00a8f6b40b59ca0115f776a60R1469-R1480

had check handle, primary key should be same as handle if it's clustered index

Reply:
... it seems, there are no indexInfo named Primary when table is PKIsHandle after debug

for example:

`create table t4(id int primary key clustered, k int, v int, unique key uk1(k))`

`t.indices` only contain `uk1`'s info